spec/rspec/mocks/test_double_spec.rb

```diff
@@ -48,6 +48,17 @@ module Mocks
           end
         end
       end
+
+      pending "test doubles able to register themselves with ObjectInspector" do
+        allow_any_instance_of(TestDouble).to receive(:name).and_return("FooBar")
+        double('User')
+
+        RSpec::Support::ObjectInspector.register TestDouble do |user|
+          "User (#{user.name})"
+        end
+
+        expect(double.inspect).to eq("User (FooBar)")
```

Comment:
This is still wrong for a spec, as it actually implements it, so the spec should just be:

```Ruby
it 'registers itself with the ObjectInspector' do
  expect(double('FooBar').inspect).to eq 'Double (FooBar)'
end
```
and then the implementation of `TestDouble` needs to be modified to call:

```Ruby
RSpec::Support::ObjectInspector.register TestDouble,  do |dbl|
  "Double (#{dbl.instance_variable_get('@name')})"
end
```

Although I'd argue it'd be nicer if the `ObjectInspector` could be registered with instances, then we could register specific instances when creating the doubles.